### PR TITLE
feat(website): Add a TRAP presentation page

### PR DESCRIPTION
**Pull request by ussefhajji**
TRAP means: "Table Ronde en Acoustique Passive". TRAPs are regular presentations given within the team on current projects, theses, etc. They are given by members of the team or external speakers.

The development here are made to add a page within OSmOSE website to list these presentations publicly.

### CODE REVIEW

File: backend/osmosewebsite/models/trap.py
Comment (ElodieENSTA) [inline backend/osmosewebsite/models/trap.py:20]: How is it used?

File: backend/osmosewebsite/models/trap.py
Comment (ElodieENSTA) [inline backend/osmosewebsite/models/trap.py:26]: With this indent, the Meta information doesn't have any use. The Meta class is used to specify how the model should be represented.

File: backend/osmosewebsite/migrations/0022_alter_trap_options.py
Comment (ElodieENSTA) [inline backend/osmosewebsite/migrations/0022_alter_trap_options.py:9]: The 0021_auto_20240702_1039 migration is missing

File: backend/osmosewebsite/admin/trap.py
Comment (ElodieENSTA) [inline backend/osmosewebsite/admin/trap.py:11]: Title should come first, that's what describe the best the TRAP

File: backend/osmosewebsite/admin/trap.py
Comment (ElodieENSTA) [inline backend/osmosewebsite/admin/trap.py:17]: We could add firstname and lastname

File: backend/osmosewebsite/management/commands/seed.py
Comment (ElodieENSTA) [inline backend/osmosewebsite/management/commands/seed.py:12]: Why do you import twice the same classes? Trap should be added to the initial import line

File: backend/osmosewebsite/management/commands/seed.py
Comment (ElodieENSTA) [inline backend/osmosewebsite/management/commands/seed.py:113]: Did you test it? News doesn't have firstname or lastname!

File: backend/osmosewebsite/management/commands/seed.py
Comment (ElodieENSTA) [inline backend/osmosewebsite/management/commands/seed.py:151]: Did you test it? TRAP doesn't have other_authors

File: backend/osmosewebsite/management/commands/seed.py
Comment (ElodieENSTA) [inline backend/osmosewebsite/management/commands/seed.py:143]: Why did you remove this code??

File: backend/osmosewebsite/serializers/trap.py
Comment (ElodieENSTA) [inline backend/osmosewebsite/serializers/trap.py:3]: Be careful to unused imports!

File: backend/osmosewebsite/tests/fixtures/trap.yaml
Comment (ElodieENSTA) [inline backend/osmosewebsite/tests/fixtures/trap.yaml:8]: Be careful! TRAP doesn't have body fields!

File: backend/osmosewebsite/tests/fixtures/trap.yaml
Comment (ElodieENSTA) [inline backend/osmosewebsite/tests/fixtures/trap.yaml:11]: Did you test it? This indent is wrong

File: backend/osmosewebsite/urls.py
Comment (ElodieENSTA) [inline backend/osmosewebsite/urls.py:3]: Should be included in views import below

File: frontend/index.html
Comment (ElodieENSTA) [inline frontend/index.html:1]: Why did you changed this file? What does your changes bring?

File: frontend/tsconfig.json
Comment (ElodieENSTA) [inline frontend/tsconfig.json:1]: What does your changes bring?

File: frontend/vite.config.ts
Comment (ElodieENSTA) [inline frontend/vite.config.ts:1]: Avoid adding this ts-ignore comments, it is really not recommended

File: frontend/vite.config.ts
Comment (ElodieENSTA) [inline frontend/vite.config.ts:17]: Why did you remove the /app ?

File: frontend/vite.config.ts
Comment (ElodieENSTA) [inline frontend/vite.config.ts:25]: This can be removed

File: package.json
Comment (ElodieENSTA) [inline package.json:1]: These libraries has nothing to do at this level! These should be removed (and the package-lock should be updated accordingly)

File: frontend/src/view/global-components/skeleton/skeleton.component.tsx
Comment (ElodieENSTA) [inline frontend/src/view/global-components/skeleton/skeleton.component.tsx:1]: No! The one from the master was the correct one! Keep in mind things can evolve while you are working. Your version on the branch you're working is not necessary the most updated one

File: frontend/src/view/global-components/authenticated-route.component.tsx
Comment (ElodieENSTA) [inline frontend/src/view/global-components/authenticated-route.component.tsx:11]: Already told you in the previous review that this will call the useEffect constantly, so this dependency array should be empty. We only need the effect to apply at component load!

File: website/src/components/Layout/Header/index.tsx
Comment (ElodieENSTA) [inline website/src/components/Layout/Header/index.tsx:34]: Why did you need this <li> with those classes?
The result is different for this link than for projects and publications

File: website/src/components/Layout/Header/index.tsx
Comment (ElodieENSTA) [inline website/src/components/Layout/Header/index.tsx:35]: you forgot some classes here to have the same style as projects and publications

File: website/src/components/Layout/Header/index.tsx
Comment (ElodieENSTA) [inline website/src/components/Layout/Header/index.tsx:58]: Why did you add this link??

File: website/src/index.css
Comment (ElodieENSTA) [inline website/src/index.css:10]: Why did you change this color? This change reflects on other pages, this must be discussed before applied!

File: website/src/pages/Trap/TrapPage.css
Comment (ElodieENSTA) [inline website/src/pages/Trap/TrapPage.css:1]: Be careful! As I already told you, class names should always be preceded by the page id, otherwise this css could be applied in an other page!

File: website/src/pages/Trap/TrapPage.tsx
Comment (ElodieENSTA) [inline website/src/pages/Trap/TrapPage.tsx:37]: I thought the title has changed?

**Review by ElodieENSTA — CHANGES_REQUESTED** [review]
The 0021 migration for osmosewebsite package is missing. I wait for this file to review the migrations.

Globally I would really appreciate that you keep all my recent updates on the master!
Keep in mind that you were working on the backend/osmosewebsite and the website folders. So you should not have changes in frontend nor backend/api! Almost all the changes in frontend override the more recent updates of the master. I will review more after you do a real look on what really is an update and what is not. Remember you work on TRAP starts after APLOSE home page, and I had to do minor changes on the APLOSE home page. So this is normal that there is some differences between your version and the master on that code.

According to the errors I saw on the seedding and the fixtures, I did not review the TRAP test file for now.

I didn't really review the TRAP page on website for now because I can't test it without the correct migrations and seeding.

For a better understanding of what you do, don't hesitate to compare with the actual version of the master, the version of your branch before the merge, and these differences between your branch and the master. Try to do a quick review here of your own changes after push. It could make you see some errors you made.

File: backend/osmosewebsite/models/trap.py
Comment (ussefhajji) [reply backend/osmosewebsite/models/trap.py:13]: The first name and last name are specifically for the presenter not for the  TRAP 
Regarding the default value for the first name, it was implemented in response to a database constraint that generated an error when a value was not provided

File: backend/osmosewebsite/models/trap.py
Comment (ussefhajji) [reply backend/osmosewebsite/models/trap.py:18]: It is used for the presenter's first name and last name.

File: backend/osmosewebsite/migrations/0022_alter_trap_options.py
Comment (ussefhajji) [reply backend/osmosewebsite/migrations/0022_alter_trap_options.py:9]: I added them in the second commit

File: backend/osmosewebsite/management/commands/seed.py
Comment (ussefhajji) [reply backend/osmosewebsite/management/commands/seed.py:112]: I didn't touch the news, that was before

File: backend/osmosewebsite/management/commands/seed.py
Comment (ussefhajji) [reply backend/osmosewebsite/management/commands/seed.py:143]: I added it

File: backend/osmosewebsite/tests/fixtures/trap.yaml
Comment (ussefhajji) [reply backend/osmosewebsite/tests/fixtures/trap.yaml]: I deleted it

File: backend/osmosewebsite/tests/fixtures/trap.yaml
Comment (ussefhajji) [reply backend/osmosewebsite/tests/fixtures/trap.yaml]: I fixed it, and it works well

File: website/src/components/Layout/Header/index.tsx
Comment (ussefhajji) [reply website/src/components/Layout/Header/index.tsx]: I deleted it

File: website/src/components/Layout/Header/index.tsx
Comment (ussefhajji) [reply website/src/components/Layout/Header/index.tsx]: I deleted it

File: website/src/index.css
Comment (ussefhajji) [reply website/src/index.css]: I put it back as before

File: website/src/pages/Trap/TrapPage.tsx
Comment (ussefhajji) [reply website/src/pages/Trap/TrapPage.tsx:37]: The team didn't suggest anything

File: backend/osmosewebsite/models/trap.py
Comment (ussefhajji) [reply backend/osmosewebsite/models/trap.py:19]: It's for the presenter not for Trap page

File: backend/osmosewebsite/models/trap.py
Comment (ElodieENSTA) [reply backend/osmosewebsite/models/trap.py:13]: I know, it was a dumb question. As I told you, the name of this field should reflect your explanation. The name of the fields, classes, functions, etc should be cristal clear to facilitate the code comprehension.
All the field that describe the presenter must be renamed.

Keeping incorrect code because there is an error is never an appropriate answer. Either the field doesn't need to be mandatory so there is no need for default value (as for lastname in fact), either we find an other mean to handle migration so this mandatory field without default value is not a problem.

File: backend/osmosewebsite/models/trap.py
Comment (ElodieENSTA) [reply backend/osmosewebsite/models/trap.py]: After our exchange, it has been removed

File: backend/osmosewebsite/models/trap.py
Comment (ElodieENSTA) [reply backend/osmosewebsite/models/trap.py:18]: Same answer as previous comment, it must be renamed

File: backend/osmosewebsite/models/trap.py
Comment (ElodieENSTA) [reply backend/osmosewebsite/models/trap.py]: The reprensentation of the model you set on this class Meta was good, you didn't have to remove it. Put it back on the Trap but with correct indent

File: backend/osmosewebsite/urls.py
Comment (ElodieENSTA) [reply backend/osmosewebsite/urls.py:3]: Nothing has changed here @ussefhajji Can you move the import of TrapViewSet in the imports below?

File: website/src/components/Layout/Header/index.tsx
Comment (ElodieENSTA) [reply website/src/components/Layout/Header/index.tsx]: Why did you delete it? We still need to have a trap link in the header. It's just the css classes that were incorrect

File: website/src/pages/Trap/TrapPage.css
Comment (ElodieENSTA) [reply website/src/pages/Trap/TrapPage.css:1]: @ussefhajji you still need to add page id before all your class css selector to avoid the css to be applied elsewhere

File: website/src/pages/Trap/TrapPage.tsx
Comment (ElodieENSTA) [reply website/src/pages/Trap/TrapPage.tsx:37]: Still need to find the correct name

File: backend/osmosewebsite/tests/views/trap.py
Comment (ElodieENSTA) [inline backend/osmosewebsite/tests/views/trap.py:22]: The test isn't executed.
Be really careful with the indent in python. Here the function is outside the class, so it won't be executed.

**Review by ElodieENSTA — CHANGES_REQUESTED** [review]

File: backend/osmosewebsite/models/trap.py
Comment (ussefhajji) [reply backend/osmosewebsite/models/trap.py:13]: I changed the name of firstname/lastname to lastname_presenter/firstname_presenter.

File: backend/osmosewebsite/models/trap.py
Comment (ussefhajji) [reply backend/osmosewebsite/models/trap.py:18]: I changed the name of linkedin_url to linkedin_url_presenter

File: website/src/pages/Trap/TrapPage.css
Comment (ussefhajji) [reply website/src/pages/Trap/TrapPage.css:1]: I changed them

File: backend/osmosewebsite/models/trap.py
Comment (ElodieENSTA) [reply backend/osmosewebsite/models/trap.py]: Thanks, no same for the other fields that concern the presenter please

File: backend/osmosewebsite/models/trap.py
Comment (ElodieENSTA) [reply backend/osmosewebsite/models/trap.py]: Be careful the change hasn't been made in the fixtures, tests and in the react code (this appy for linkedin_url and the other field regarding presenter)

File: backend/osmosewebsite/models/trap.py
Comment (ElodieENSTA) [reply backend/osmosewebsite/models/trap.py:23]: The indent is still incorrect, the Meta class should be inside the Trap class

File: website/src/components/Layout/Header/index.tsx
Comment (ElodieENSTA) [reply website/src/components/Layout/Header/index.tsx]: Some classes are still missing on the trap link

**Comment by ixio** [comment]
Long time no see, just popping in for a random check. All 10 migration files here are related to the new trap model right ? You should delete all those migrations files and re-run makemigrations in order to have a single clean migration file :wink:.

Also I have no idea what TRAP is :sweat_smile: a bit of a shame it's not documented in the code, the PR description or name. Maybe you were waiting when merging the PR to link in the merge commit name a JIRA ticket that has the relevant info ? Such info is useful for future generations that will also work on the project :wink:

**Comment by ElodieENSTA** [comment]
> Long time no see, just popping in for a random check. All 10 migration files here are related to the new trap model right ? You should delete all those migrations files and re-run makemigrations in order to have a single clean migration file 😉.

Just made a commit to clean that, thanks

> Also I have no idea what TRAP is 😅 a bit of a shame it's not documented in the code, the PR description or name. Maybe you were waiting when merging the PR to link in the merge commit name a JIRA ticket that has the relevant info ? Such info is useful for future generations that will also work on the project 😉


Yes I understand, I edited the message of the PR. I'm waiting for a definitive name for the page (because TRAP is indeed not really understandable) to maybe rename all of the TRAP mentions in the code

**Comment by ElodieENSTA** [comment]
The team has chosen the name "Scientific talks" for this page

**Review by ElodieENSTA — APPROVED** [review]

**Pull request merged**
